### PR TITLE
User Inheritance

### PR DESCRIPTION
**Pull request by massivesmo**
Enhancements to add the possibility to inherit from BaseUser class.

### CODE REVIEW

**Comment by danrot** [comment]
@massivesmo For what did you introduce the `BaseUserRole`-class?

**Comment by massivesmo** [comment]
The BaseUser class was introduced to make it inheritable.
For a special case we need to make the UserRole specific to an other Object. This means to add a field in this table.

File: src/Sulu/Bundle/SecurityBundle/Entity/BaseUserRole.php [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+ * UserRole
+ * @ExclusionPolicy("all");
+ */
+abstract class BaseUserRole extends ApiEntity
+{
+    /**
+     * @var integer
+     * @Expose
+     */
+    private $id;
```
Comment (danrot) [inline src/Sulu/Bundle/SecurityBundle/Entity/BaseUserRole.php]: Does this work? Shouldn't these properties also be protected?

Comment (massivesmo) [inline src/Sulu/Bundle/SecurityBundle/Entity/BaseUserRole.php]: This is working on our current project.

**Pull request merged**